### PR TITLE
add non facet type and update csv export

### PR DESCRIPTION
**Pull request by qgan7125**

### CODE REVIEW

File: src/api/server.js [hunk 1]
```diff
@@ -70,19 +70,11 @@ server.submitSuggestQuery = (suggestQuery, callback) => {
...
-    headers: {
-      "Content-type": "application/x-www-form-urlencoded",
-      ...(query.userpass ? {"Authorization": "Basic " + query.userpass} : {}),
-    }
-  }, (err, resp) => {
+
+  const queryString = solrQuery({...query, rows: MAX_INT}, {wt: "csv"});
+  const options = getXHROptions(query, queryString);
+
+  server.performXhr(options, (err, resp) => {
```
Comment (qgan7125) [inline src/api/server.js:77]: Update the way to query `csv` data.

File: src/components/component-pack.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ export default {
     "range-facet": RangeFacet,
     "period-range-facet": RangeFacet,
     "date-range-facet": RangeFacet,
+    "non-facet": TextSearch,
```
Comment (qgan7125) [inline src/components/component-pack.js]: Set the default `non-facet` as `TextSearch` facet

File: src/api/solr-query.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ const textFieldToQueryFilter = (field) => {
 };
 
 const fieldToQueryFilter = (field) => {
-  if (field.type === "text" && field.field !== "*") {
+  if ((field.type === "text" || field.type === "non-facet" )&& field.field !== "*") {
```
Comment (qgan7125) [inline src/api/solr-query.js]: Add `non-facet` to the query string

File: src/components/results/csv-export.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ export default function (props) {
       "pull-right": bootstrapCss,
       "btn-xs": bootstrapCss
     })}>
-      Export excel
+      Export csv
```
Comment (qgan7125) [inline src/components/results/csv-export.js:13]: rename the button name.

**Review by datadavev — APPROVED** [review]

File: src/api/server.js [hunk 5]
```diff
@@ -70,19 +70,11 @@ server.submitSuggestQuery = (suggestQuery, callback) => {
...
-      "csv.separator": ";"
-    }),
-    method: "POST",
-    headers: {
-      "Content-type": "application/x-www-form-urlencoded",
-      ...(query.userpass ? {"Authorization": "Basic " + query.userpass} : {}),
-    }
-  }, (err, resp) => {
+
+  const queryString = solrQuery({...query, rows: MAX_INT}, {wt: "csv"});
```
Comment (dannymandel) [inline src/api/server.js:74]: Why don't we want the separators specified anymore?

File: src/api/solr-query.js (hunk 3 above)
Comment (dannymandel) [reply src/api/solr-query.js]: What does `non-facet` do?

File: src/api/server.js (hunk 5 above)
Comment (qgan7125) [reply src/api/server.js:74]: The `separator` seems only to change the separators of the result values but the names of the columns still use commas. When I imported the download files with those separators to excel, Excel would not format records reasonably even though we set the different delimiters. After I removed separators, the `csv` files work well. In addition, I found solr could return `xlsx` format results. I will explore it and check if it works.

**Review by dannymandel — APPROVED** [review]
LGTM, thanks @qgan7125!

**Pull request merged**